src/FormContainer.tsx

```diff
@@ -16,8 +16,17 @@ const makeWrapper = <T extends {}>(config: IFormConfig<T>) => (WrappedComponent:
             };
         }
 
-        setModel = (model: { [name in keyof T]: any }) => {
-            this.setState({ model });
+        setModel = (model: { [name in keyof T]: any }, setTouched: boolean = false) => {
```

Comment:
this method has one single responsibility to set the model. I think `setTouched` shouldn't reside in this method.